### PR TITLE
[formrecognizer] adds AsyncLROPoller and continuation token support

### LINKED ISSUE #11530: [formrecognizer] Update async methods to return AsyncLROPoller and support continuation token
Currently waiting on:
1. PR for azure-core: https://github.com/Azure/azure-sdk-for-python/pull/10801
2. Work from autorest to support this, which is pending.

This will allow us to return `AsyncLROPoller` for our async methods and support continuation tokens. Because of the change in return type we will do a re-naming to align with other languages (feel free to spin-off on a different issue):

* sync method `begin_train_model` -> `begin_training`
* async method `train_model` -> `begin_training`
* all async methods will now have `begin` prefix added to them

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves #11530 

**TODO:**

* [x] Update tests
* [x] Update samples
* [x] Update README
* [x] Update shared requirements
* [x] Update Changelog
* [x] Live tests for python 2.7 are failing due to an un-pickle-able component of the logger added by pytest. Wait for fix: https://github.com/Azure/azure-sdk-for-python/issues/11686
* [x] wait for azure-core 1.6.0 to release

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 1]
```diff
@@ -4,13 +4,33 @@
...
+- Sync method `begin_train_model` renamed to `begin_training`
+- `training_files` parameter of `begin_training` is renamed to `training_files_url`
+- `use_labels` parameter of `begin_training` is renamed to `use_training_labels`
 - `list_model_infos` method has been renamed to `list_custom_models`
 - Removed `get_form_training_client` from `FormRecognizerClient`
 - Added `get_form_recognizer_client` to `FormTrainingClient`
-- A `HttpResponseError` is now raised if a model with `status=="invalid"` is returned from the `begin_train_model()` or `train_model()` methods
+- A `HttpResponseError` is now raised if a model with `status=="invalid"` is returned from the `begin_training` methods
 - `PageRange` is renamed to `FormPageRange`
+<<<<<<< HEAD
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md]: Seems that a merge conflict was not resolved here

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +51,10 @@
   - see the [Azure Identity documentation](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/identity/azure-identity/README.md) for more information
 - `page_number` attribute has been added to `FormTable`
 
+**Dependency updates**
+
+- Adopted [azure-core](https://pypi.org/project/azure-core/) version 1.6.0 or greater
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:46]: Should we add this to the setup.py now, or are you waiting until azure-core actually gets released to add it

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ def __init__(self, endpoint, credential, **kwargs):
         authentication_policy = get_authentication_policy(credential)
         self._client = FormRecognizer(
             endpoint=endpoint,
-            credential=credential,
+            credential=credential,  # type: ignore
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:69]: What is the typing issue here?

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 4]
```diff
@@ -157,6 +161,7 @@ def begin_recognize_receipts_from_url(self, receipt_url, **kwargs):
         """
 
         polling_interval = kwargs.pop("polling_interval", POLLING_INTERVAL)
+        continuation_token = kwargs.pop("continuation_token", None)
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:164]: Since you're just popping continuation_token and putting it back in, you should just not pop it at all, as the kwargs passed into the function call will have it

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py [hunk 5]
```diff
@@ -122,8 +123,23 @@ def begin_train_model(self, training_files_url, use_training_labels, **kwargs):
...
+            model = self._client._deserialize(Model, raw_response)
+            return CustomFormModel._from_generated(model)
+
         cls = kwargs.pop("cls", None)
+        continuation_token = kwargs.pop("continuation_token", None)
         polling_interval = kwargs.pop("polling_interval", POLLING_INTERVAL)
+        deserialization_callback = cls if cls else callback
+
+        if continuation_token:
+            return LROPoller.from_continuation_token(
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:136]: How come `begin_train_model` calls LROPoller.from_continuation_token, while the other polling methods do not call it if a continuation token is present

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py [hunk 6]
```diff
@@ -85,19 +85,19 @@ def __init__(
...
 
     @distributed_trace_async
-    async def train_model(
+    async def begin_training(
             self,
             training_files_url: str,
             use_training_labels: bool,
             **kwargs: Any
-    ) -> CustomFormModel:
+    ) -> AsyncLROPoller:
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py:100]: I believe latest azure-core will allow you to do AsyncLROPoller[CustomFormModel] in typing.

Same goes for the sync pollers, but that can be another PR

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py [hunk 7]
```diff
@@ -146,16 +167,15 @@ async def train_model(
...
-            return CustomFormModel._from_generated(model)
-
-        deserialization_callback = cls if cls else callback
-        return await async_poller(
+        return AsyncLROPoller(
             self._client._client,
             response,
             deserialization_callback,
-            AsyncLROBasePolling(timeout=polling_interval, lro_algorithms=[TrainingPolling()], **kwargs)
+            AsyncLROBasePolling(  # type: ignore
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py:174]: Why is there a type ignore comment here?

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md (hunk 2 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:46]: it is added to setup.py

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py (hunk 3 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:69]: error: Argument "credential" to "FormRecognizerClient" has incompatible type "Union[AzureKeyCredential, TokenCredential]"; expected "TokenCredential"

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py (hunk 4 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:164]: I'd like to do that but in many of the methods I'm controlling the lro_algorithm so I need to pass in, e.g., LROBasePolling(**kwargs) myself into the generated call. I need to pop it here so it doesn't get passed into the polling method.

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py (hunk 5 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:136]: This one doesn't generate as an LRO (although it is an LRO). It uses Location polling which should not be used for POST/201, from my understanding. I believe there is an issue open in autorest to handle this, but for now I have to work around it like this. The other methods do call it if continuation_token is present, but in the generated code.

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py (hunk 7 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py:174]: error: Argument 4 to "AsyncLROPoller" has incompatible type "AsyncLROBasePolling"; expected "AsyncPollingMethod[Any]"

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py (hunk 6 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py:100]: Cool! Let me try
